### PR TITLE
NO-ISSUE: api: add support for resource monitor configuration

### PR DESCRIPTION
**Pull request by hexfusion**
This PR exposes the configuration for resource monitors. If no monitors are defined they will be disabled. Each resource can define one alert rule per severity which for now is `critical`,  `warning`, and `info` this verification will need to be done by the service.

 ```yaml
resources:
  - monitorType: cpu
    alertRules:
     - severity: Warning
      duration: 10m
      percentage: 50
      description: 'CPU load is above 50% for more than 10 minutes.'
    samplingInterval: 2s
  - monitorType: memory
    alertRules:
     - severity: warning
      duration: 5m
      percentage: 80
      description: 'Memory usage is above 80% for more than 5 minutes'
    samplingInterval: 1s
  - monitorType: disk
    alertRules:
    - severity: warning
      duration: 5h
      percentage: 80
      description: 'Disk usage is above 80% for more than 5 hours.'
    samplingInterval: 2m
    path: /
  - monitorType: battery
    alertRules:
    - severity: Critical
      duration: 1m
      percentage: 5
      description: string
    samplingInterval: 1s
```

### CODE REVIEW

File: api/v1alpha1/openapi.yaml [hunk 1]
```diff
@@ -2529,6 +2632,22 @@ components:
               type: array
               items:
                 type: string
+        resources:
+          type: object
```
Comment (fzdarsky) [inline api/v1alpha1/openapi.yaml:2635]: Could we make this an array to eventually allow user-defined resource monitors (e.g. battery level)?

File: api/v1alpha1/openapi.yaml [hunk 2]
```diff
@@ -2529,6 +2631,22 @@ components:
               type: array
               items:
                 type: string
+        resources:
```
Comment (celdrake) [inline api/v1alpha1/openapi.yaml:2634]: Just to confirm, we'd keep the `device.status.resources` with the summary status for each device resource.
And additionally, this new `device.spec.resources` would have the alarm details.

LGTM.

**Review by celdrake — APPROVED** [review]

File: api/v1alpha1/openapi.yaml [hunk 3]
```diff
@@ -2529,6 +2631,22 @@ components:
...
+          type: object
+          properties:
+            cpu:
+              $ref: '#/components/schemas/ResourceCPUConfigSpec'
```
Comment (fzdarsky) [inline api/v1alpha1/openapi.yaml:2638]: The resources and alertRules are essentially the same, right? Could we have one generic ResourceAlertRule schema that applies to all of them (plus user-defined ones)?

File: api/v1alpha1/openapi.yaml (hunk 1 above)
Comment (hexfusion) [reply api/v1alpha1/openapi.yaml:2635]: so like oneOf

 type: array
  description: 'Array of resource monitor configurations.'
  items:
    type: object
    oneOf:
      - $ref: '#/components/schemas/ResourceCPUConfigSpec'
      - $ref: '#/components/schemas/ResourceMemoryConfigSpec'
      - $ref: '#/components/schemas/ResourceDiskConfigSpec'
    discriminator:
      propertyName: type
      mapping:
        cpu: '#/components/schemas/ResourceCPUConfigSpec'
        memory: '#/components/schemas/ResourceMemoryConfigSpec'
        disk: '#/components/schemas/ResourceDiskConfigSpec'

Comment (hexfusion) [reply api/v1alpha1/openapi.yaml:2635]: will play with it

File: api/v1alpha1/openapi.yaml [hunk 4]
```diff
@@ -1975,6 +1975,67 @@ components:
...
+          type: string
+        alertRules:
+          type: array
+          items:
+            $ref: '#/components/schemas/ResourceAlertRule'
+          description: "Array of alert rules. Only one alert per severity is allowed." 
+        frequency:
+          type: string
+          pattern: '^\d+[smhd]$'
+          description: "Frequency is the time between disk usage checks. Format: number followed by 's' for seconds, 'm' for minutes, 'h' for hours, 'd' for days."
```
Comment (hexfusion) [inline api/v1alpha1/openapi.yaml:1991]: nit

File: api/v1alpha1/openapi.yaml (hunk 1 above)
Comment (hexfusion) [reply api/v1alpha1/openapi.yaml]: I guess I don't understand how the user can update the api. You would need something like CRD right?

Comment (hexfusion) [reply api/v1alpha1/openapi.yaml]: Any monitor would require agent support so I'm not sure I understand how a generic API would be helpful the type would need to be extended based on support. Maybe you can provide a more concrete example.

Comment (hexfusion) [reply api/v1alpha1/openapi.yaml]: per conversation this would be future support

**Pull request merged**